### PR TITLE
Validation for Position and Rotation Coords Issue#1290

### PR DESCRIPTION
**Pull request by hridpath**
Overview
UnityEngine.XR.WSA   InteractionSourceState.sourcePose.TryGetPosition and InteractionSourceState.sourcePose.TryGetRotation can return 'NaN' and or 'Infinity/NegativeInfinty' as a coordinate value. When setting a transform.postion or transform.rotation with non numeric coord values will through an error.

Changes
Added two validation methods to test for 'IsNaN' and 'IsInfinity'. Applied these changes to the condition statement befor setting prosition and rotation of controller model.
- Fixes: #1290.

### CODE REVIEW

File: Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs [hunk 1]
```diff
@@ -157,23 +157,35 @@ private void UpdateControllerState()
                     }
 
                     Vector3 newPosition;
-                    if (sourceState.sourcePose.TryGetPosition(out newPosition, InteractionSourceNode.Grip))
+                    if (sourceState.sourcePose.TryGetPosition(out newPosition, InteractionSourceNode.Grip) && ValidPosition(newPosition)) // issue #1290
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: please remove the comment

**Comment by hridpath** [comment]
These have been removed

File: Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs [hunk 2]
```diff
@@ -157,23 +157,33 @@ private void UpdateControllerState()
...
+                    if (sourceState.sourcePose.TryGetRotation(out newRotation, InteractionSourceNode.Grip) && ValidRotation(newRotation))
                     {
                         currentController.ControllerParent.transform.localRotation = newRotation;
                     }
                 }
             }
 #endif
         }
-
+		private bool ValidRotation(Quaternion newRotation)
```
Comment (keveleigh) [inline Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: Please double check tabs -> spaces :)

Comment (hridpath) [reply Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: I am using the default VS 2017 formatting. What should I use?

Comment (hridpath) [reply Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: Tab size is 4

Comment (keveleigh) [reply Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: Yep, the Toolkit uses 4 spaces as a tab. Several of the added lines in here (such as the one this comment is on) are actual tabs though.

Comment (hridpath) [reply Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs]: Found it. Tabify is on.

**Pull request merged**